code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,177 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging.contentcards
+
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.SmallImageUI
+import com.adobe.marketing.mobile.aepcomposeui.state.SmallImageCardUIState
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepButton
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepDismissButton
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepImage
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepText
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.SmallImageTemplate
+import com.adobe.marketing.mobile.messaging.ContentCardSchemaData
+import com.adobe.marketing.mobile.messaging.MessagingConstants
+import com.adobe.marketing.mobile.messaging.Proposition
+import com.adobe.marketing.mobile.messaging.SchemaType
+import com.adobe.marketing.mobile.util.DataReader
+
+/**
+ * Utility object for creating instances of AEP model classes from Map data structures.
+ * This utility handles parsing and transforming content card data into the relevant AEP UI models,
+ * such as [AepText], [AepImage], [AepButton], and [AepDismissButton], as well as creating
+ * the corresponding UI templates like [SmallImageTemplate].
+ */
+internal object ContentCardSchemaDataUtils {
+    /**
+     * Creates an instance of [AepText] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepText].
+     * @return An instance of [AepText].
+     */
+    private fun createAepText(map: Map<String, Any>): AepText {
+        return AepText(
+            content = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, "")
```

Comment:
We're not really enforcing `content` being mandatory for `AepText` here. Would you please add that check? Maybe something like this?
```
DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, null)?.let {
     AepText(content = it)
}
```